### PR TITLE
Upgrade Kotlin version and use more idiomatic Kotlin

### PR DESCRIPTION
**Pull request by mlvandijk**
## Summary

Upgrade to Kotlin v1.3.0
Use more idiomatic Kotlin in the examples.

## Details

Use [data class](https://kotlinlang.org/docs/reference/data-classes.html)
Remove redundant {} and ->

## Motivation and Context
Update & make it nicer

## How Has This Been Tested?
mvn clean install

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
Upgrade / refactor

## Checklist:

- [ ] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
N/a

### CODE REVIEW

File: kotlin-java8/src/test/kotlin/cucumber/runtime/kotlin/test/TypeRegistryConfiguration.kt [hunk 1]
```diff
@@ -15,13 +15,10 @@ class TypeRegistryConfiguration : TypeRegistryConfigurer {
...
-                LambdaStepdefs.Person::class.java,
-                TableEntryTransformer<LambdaStepdefs.Person>
+                Person::class.java,
+                TableEntryTransformer<Person>
                 { map: Map<String, String> ->
-                    val person = LambdaStepdefs.Person()
-                    person.first = map.get("first")
-                    person.last = map.get("last")
-                    person
+                    Person(map["first"], map["last"])
```
Comment (mpkorstanje) [inline kotlin-java8/src/test/kotlin/cucumber/runtime/kotlin/test/TypeRegistryConfiguration.kt]: :heart:

File: kotlin-java8/src/test/kotlin/cucumber/runtime/kotlin/test/LambdaStepdefs.kt [hunk 2]
```diff
@@ -62,9 +62,6 @@ class LambdaStepdefs : En {
...
         }
     }
-
-    class Person {
-        var first: String? = null
-        var last: String? = null
-    }
 }
+
+data class Person(val first: String?, val last: String?)
```
Comment (mpkorstanje) [inline kotlin-java8/src/test/kotlin/cucumber/runtime/kotlin/test/LambdaStepdefs.kt]: Please do add a new line.

File: kotlin-java8/src/test/kotlin/cucumber/runtime/kotlin/test/RunCukesTest.kt [hunk 3]
```diff
@@ -4,5 +4,4 @@ import cucumber.api.junit.Cucumber
 import org.junit.runner.RunWith
 
 @RunWith(Cucumber::class)
-class RunCukesTest {
-}
+class RunCukesTest
```
Comment (mpkorstanje) [inline kotlin-java8/src/test/kotlin/cucumber/runtime/kotlin/test/RunCukesTest.kt]: Please do add a new line.

File: kotlin-java8/src/test/kotlin/cucumber/runtime/kotlin/test/TypeRegistryConfiguration.kt (hunk 1 above)
Comment (mlvandijk) [reply kotlin-java8/src/test/kotlin/cucumber/runtime/kotlin/test/TypeRegistryConfiguration.kt]: Right? ;)

**Pull request merged**